baggage/baggage.go

```diff
@@ -220,26 +237,40 @@ func (p properties) String() string {
 type Member struct {
 	key, value string
 	properties properties
+
+	// hasData indicates whether the created property contains data or not.
+	// Properties that do not contain data are invalid with no other check
+	// required.
+	hasData bool
 }
 
 // NewMember returns a new Member from the passed arguments. An error is
 // returned if the created Member would be invalid according to the W3C
 // Baggage specification.
 func NewMember(key, value string, props ...Property) (Member, error) {
-	m := Member{key: key, value: value, properties: properties(props).Copy()}
+	m := Member{
+		key:        key,
+		value:      value,
+		properties: properties(props).Copy(),
+		hasData:    true,
+	}
 	if err := m.validate(); err != nil {
-		return Member{}, err
+		return newInvalidMember(), err
 	}
 
 	return m, nil
 }
 
+func newInvalidMember() Member {
+	return Member{}
+}
+
```

Comment:
Same question here regarding if this is needed?

Reply:
Same as for the property. It's not needed per-se, it just seems more readable to me.